tools/coverage.sh

```diff
@@ -69,18 +69,43 @@ export RUSTFLAGS="$RUSTFLAGS -Z unstable-options -Cinstrument-coverage -Zlinker-
 export RUSTDOCFLAGS="$RUSTDOCFLAGS -C instrument-coverage -Z unstable-options --persist-doctests target/debug/doctestbins -Zlinker-features=-lld"
 export LLVM_PROFILE_FILE='conjure-oxide-%p-%m.profraw' 
 
+# regex patterns to ignore
+GRCOV_EXCLUDE_LINES=(
+  'consider covered'
+  'bug!'
+  '#\[register_rule'
+  'register_rule_set!'
+  'unreachable!'
```

Comment:
instead of unreachable I suggest we use bug! so it produces the standardised error message. if so, since we won't encourage its use unreacahable can be removed from this list.